extensions/interactions/AlgebraicExpressionInput/protractor.js

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview End-to-end testing utilities for Algebraic Expression Input
+ * interaction
+ */
+
+var objects = require(process.cwd() + '/extensions/objects/protractor.js');
+
+var customizeInteraction = function() {
+  // There are no customizations.
+};
+
+var expectInteractionDetailsToMatch = async function(elem) {
```

Comment:
This function should also check that the interaction was created correctly. In this case we don't apply any customizations, so we should confirm that the interaction appears in its default state. Sorry about the bad examples elsewhere in the codebase. Unfortunately some of the other interactions have lax tests that we should fix at some point

Reply:
I understand. I assume we would have to check that initially, the guppy editor has empty content.
The thing with getting the value from the guppy editor is that you need the reference to the instance of the Guppy class that is formed while initializing new guppy instances. This is done in the `guppy-initialization.service.ts`. This reference also requires the ID of the element. That ID is assigned to the div while initialization and is a unique random number that we won't have access to in the protractor tests since we are not storing it anywhere. We can't simply do things like `getAttribute(value)` since the div itself stores this value in an expression tree with a bunch of KaTeX HTML elements and they get converted to the required format when we try to fetch the value from the instance that created that particular div, which again, we don't have access to in the protractor tests.

Reply:
Okay, it sounds like we need to build a more complicated infrastructure for working with these objects like we have with rich text editors, and I understand that you probably don't have time to build that right now. Instead, could we do something like count the number of KaTeX elements so that we at least know the editor changes when the user inputs something? At least then we'd be able to catch bugs where the editor doesn't respond.

Reply:
I've extracted the annotation tag from KaTeX element. That contains the XML content but since we are just testing for default state we can expect it to just be empty.

Reply:
Nice! This looks like a great solution